sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/EntityScopeFactory.cs

```diff
@@ -80,6 +82,28 @@ public static bool TryExtractDiagnosticId(IDictionary<string, object> properties
             return false;
         }
 
+        /// <summary>
+        ///   Extracts a diagnostic id from a message's properties.
+        /// </summary>
+        ///
+        /// <param name="properties">The properties holding the diagnostic id.</param>
+        /// <param name="id">The value of the diagnostics identifier assigned to the event. </param>
+        ///
+        /// <returns><c>true</c> if the event was contained the diagnostic id; otherwise, <c>false</c>.</returns>
+        ///
+        public static bool TryExtractDiagnosticId(PropertiesMap properties, out string id)
+        {
+            id = null;
+
+            if (properties.TryGetValue<string>(DiagnosticProperty.DiagnosticIdAttribute, out var objectId) && objectId is string stringId)
```

Comment:
It looks like TryGetValue will handle the type cast to string, so I think we can just do
```suggestion
            if (properties.TryGetValue<string>(DiagnosticProperty.DiagnosticIdAttribute, out string stringId))
```